lib/stackprof/report.rb

```diff
@@ -9,7 +9,7 @@ def initialize(data)
     attr_reader :data
 
     def frames(sort_by_total=false)
-      Hash[ *@data[:frames].sort_by{ |iseq, stats| -stats[sort_by_total ? :total_samples : :samples] }.flatten(1) ]
+      *@data[:frames].sort_by{ |iseq, stats| -stats[sort_by_total ? :total_samples : :samples] }.inject({}){|h, (k, v)| h[k] = v; h}
```

Comment:
Doesn't look like there's good test coverage for the reporter anyway. I should have reviewed this more carefully, will revert for now.